c7n/resources/secretsmanager.py

```
@@ -0,0 +1,136 @@
+    filter_registry = filters
+    permissions = ('secretsmanager:ListSecretVersionIds',)
+
+    class resource_type(object):
+        service = 'secretsmanager'
+        enum_spec = ('list_secrets', 'SecretList', None)
+        detail_spec = ('describe_secret', 'SecretId', 'ARN', None)
+        id = 'ARN'
+        name = 'Name'
+        dimension = None
+        filter_name = None
+
+    def augment(self, resources):
+        client = local_session(self.session_factory).client('secretsmanager')
+
+        def _augment(r):
+            r['Versions'] = client.list_secret_version_ids(SecretId=r['ARN']).get('Versions')
+            self.log.info('adding version info')
+            return r
+
+        resources = super(SecretsManager, self).augment(resources)
+
+        with self.executor_factory(max_workers=1) as w:
+            return list(filter(None, w.map(_augment, resources)))
+
+
+@SecretsManager.action_registry.register('tag')
+class TagSecretsManagerResource(Tag):
+    """Action to create tag(s) on a Secret resource
+
+    :example:
+
+    .. code-block:: yaml
+
+        policies:
+            - name: tag-secret
+              resource: secrets-manager
+              actions:
+                - type: tag
+                  key: tag-key
+                  value: tag-value
+    """
+
+    permissions = ('secretsmanager:TagResource',)
+
+    def process_resource_set(self, resources, tags):
+        client = local_session(self.manager.session_factory).client('secretsmanager')
+        for r in resources:
+            r['Tags'].extend([{'Key': t['Key'], 'Value':t['Value']} for t in tags])
+            try:
+                client.tag_resource(SecretId=r['ARN'], Tags=r['Tags'])
+            except ClientError as e:
+                if e['Error']['Code'] == 'InternalFailure':
+                    self.log.error(
+                        'Exception when tagging resource %s' %
+                        (r['ARN'], e['Error']['Code']))
```

this is silently dropping all other errors.